### ISSUE TITLE
Feature Request / Suggestion: Sorted Conatiner List

### ISSUE THREAD

**Issue by BasicBaer** [mozilla/multi-account-containers#1866]
If the list of containers becomes longer than a few entries it gets quite annoying to search for the one container you want to use.

Therefore, I would suggest sorting the entries alphabetically.

**Comment by laggingreflex** [comment]
I'd request an option to sort by "icon". 

I'd also request the sort order be customizable.

**Comment by mikhey** [comment]
It would be nice to be able to sort them in a custom order using handles, by name.  In addition, it would be an added plus have a search above the list.  Grouping would be nice as well, as for example I have several google accounts that I keep in separate tab containers.  

Maybe this is all a sign that I'm "abusing" the original design?  :-/

**Comment by 211217613** [comment]
I think a first good step would be to have an option in about:config or somewhere that lets you toggle new containers being opened in a sorted order.

So if I open a new banking container it will open to the right of the current banking container.

**Comment by bircher** [comment]
I just synced the containers to a new browser and the original order is totally messed up and there is no way to manually re-order them.
Maybe there is a secret hidden config somewhere I didn't find where the container list config can be edited?

**Comment by groovecoder** [comment]
This issue seems to overlap heavily with #330 which was fixed in #1608 and 7.3.0 release is pushed to AMO waiting for review and publish. Are people here happy with that fix for this issue, or should we keep this open?

**Comment by grahamperrin** [comment]
@groovecoder thanks for asking, there was overlap but I do think that there's value in retaining this as an enhancement request. 

My list of containers is not particularly long, but I would have preferred the option of an immediate sort (not a succession of single moves). 

This is not lack of gratitude for https://github.com/mozilla/multi-account-containers/pull/1608 `:-)` – it's a subtle but important enhancement that may be **greatly** appreciated by users whose lists are longer than mine. 

Whilst alphabetical makes most sense (to me), I recognise the other use cases.

**Comment by grahamperrin** [comment]
https://github.com/mozilla/multi-account-containers/issues/1866#issuecomment-759325649

> … synced the containers …

I sync preferences but not add-ons. 

Does excluding add-ons also exclude _preferences for_ add-ons? 

(The containers that I added to my everyday profile are missing from a test profile, which I synchronised this morning. This surprised me because whilst use of the add-on is commonplace, containers are – essentially – a feature of Firefox.)

**Comment by keitalbame** [comment]
Please add alphabetical sorting of containers. I use containers for every site I need to login.
The ability to manually sort the list(thanks for that :+1: ) but I prefer to have it alphabetically sorted.

**Comment by zpcol** [comment]
It is a pain if the container names are not automatically sorted, adding a folder will also be helpful.

**Comment by grahamperrin** [comment]
https://github.com/mozilla/multi-account-containers/issues/1866#issue-702580457

> alphabetically

#1673

**Comment by frnk01** [comment]
Why isnt the alphabetically ordering being implemented? Anyone on the team that can comment?

**Comment by enixnetworks** [comment]
Hello!  I would like to add my name to the list of people asking for a solution to this issue.  All containers are all sorted correctly in the add-on window, but those changes do not carry over to the right click context menu.  Other than this little hiccup, this is a great app.  I use it frequently throughout the day.  

![snip1](https://user-images.githubusercontent.com/31581348/127183063-afe8f4fb-d066-49bc-8a9d-30d5b58e0ad4.PNG)
![snip2](https://user-images.githubusercontent.com/31581348/127183081-a76e4c31-8f41-4d36-b2a2-c5b7cadd3f40.PNG)

**Comment by enixnetworks** [comment]
Hi, just want to add my name to the list.  I have a pile of containers and the right click context menu is almost useless.  The add-on icon menu is sorted and much easier to use from a functionality standpoint, but the right click option would be faster if this issue was sorted.  No pun intended, of course.  Great app.  

![snip1](https://user-images.githubusercontent.com/31581348/127184645-b80ea0e2-45e4-4f68-b3c8-8c9ae68bb4f4.PNG)
![snip2](https://user-images.githubusercontent.com/31581348/127184657-8603acd0-bf0c-427c-8b40-2d97a8be7152.PNG)

**Comment by VerisZG** [comment]
BUMP - please ADD SORTING the list in the RIGHT CLICK TAB MENU (open with) according to the sorted list in container list management.

**Comment by dannycolin** [comment]
> BUMP - please ADD SORTING the list in the RIGHT CLICK TAB MENU (open with) according to the sorted list in container list management.

This can't be done in Multi-Account Container. It needs to be fixed in Firefox itself.

**Comment by TetraAsh** [comment]
Just wanted to support this idea..... i have over 100 containers, and alphabetical sorting would be amazing!

**Comment by TechRavingMad** [comment]
I want to add my support for this too.  At least alphabetical as a default, specifically the right-click "Open in..." menu.

Best option would be that whenever the list is displayed display it in whatever sort order is in the extension drop-down.

**Comment by R4RD0** [comment]
I just moved to a new device and lost my manually sorted list in the process. :(
I had 160 odd listed in there and really didn't want to be dragging and scrolling again.

So went to PowerShell to create a workaround for this and added an option to sort alphabetically by name, icon(name), colour and the default sorting for anyone wanting to sort them while we wait for a more official solution. 
https://github.com/R4RD0/Sort-Firefox-Containers

**Comment by xanoni** [comment]
Someone posted a workaround here: https://github.com/mozilla/multi-account-containers/issues/2318#issuecomment-1121943300

**Comment by VerisZG** [comment]
> I just moved to a new device and lost my manually sorted list in the process. :( I had 160 odd listed in there and really didn't want to be dragging and scrolling again.
> 
> So went to PowerShell to create a workaround for this and added an option to sort alphabetically by name, icon(name), colour and the default sorting for anyone wanting to sort them while we wait for a more official solution. https://github.com/R4RD0/Sort-Firefox-Containers

Great script!
I modified it a bit to sort by color ascending, then icon descending, then name ascending. For me it makes most sense. If anyone wanted similar functionality just change one line in the script to something like this:

$sortedIdentities = $json.identities | Sort-Object -Property @{Expression = "color"; Descending = $false}, @{Expression = "icon"; Descending = $true}, @{Expression = "name"; Descending = $false}

**Comment by xanoni** [comment]
> I had 160 odd listed in there

I think this is the main issue. The extension was designed for folks with just a handful of containers, while you can get much better isolation/privacy with 50+ containers. The UI should be revisited with that in mind.

**Comment by VerisZG** [comment]
> > I had 160 odd listed in there
> 
> I think this is the main issue. The extension was designed for folks with just a handful of containers, while you can get much better isolation/privacy with 50+ containers. The UI should be revisited with that in mind.

Nested menu might solve many problems or an option after how many items the list should be wrapped to another column. If someone created such an addon, it shouldn't be too hard to add some options to it. My list is growing bigger because I use 1 containter per client (i mean real client like a company). So the list might grow maybe to even 200 items someday.

**Comment by MarcusWolschon** [comment]
I sorted my containers using the "manage container"+"drag&drop" mechanism
but the sorting was not applied to the "open existing tab in new container" menu.

**Comment by xanoni** [comment]
> but the sorting was not applied to the "open existing tab in new container" menu.

yep, here is the workaround until it gets fixed: https://github.com/mozilla/multi-account-containers/issues/2318#issuecomment-1121943300

**Comment by tehmessiah75** [comment]
Same issue here.
I thought I had found a fix.
There is a containers.json file 
C:\Users\<USERNAME>\AppData\Roaming\Mozilla\Firefox\Profiles\<PROFILE.FOLDERS>\containers.json
and
C:\Users\<USERNAME>\AppData\Local\Mozilla\Firefox\Profiles\<PROFILE.FOLDERS>\containers.json

I resorted the entries in the file which took about 2hrs and then imported it back to firefox.
However after a restart it reset them back to their original sorting.
When I check the containers.json file it is as I saved them.
There is obviously something else that also works with this file as far as the sorting is concerned.

Like everyone else appears to have suggested sorting options.
I believe the default should be alphabetically sorted with options to sort by icon or colour (US = color).

I didn't know the sync option was available, kudos for making that a thing.
Keep up the good work, it is a great add-on.

**Comment by bugoverfl0w** [comment]
Hello Team,

Do you implement sort container by name.

Thank you,
